user_tools/src/spark_rapids_pytools/pricing/dataproc_pricing.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""providing absolute costs of resources in GCloud Dataproc"""
+
+from dataclasses import dataclass
+
+
+from spark_rapids_pytools.common.prop_manager import JSONPropertiesContainer
+from spark_rapids_pytools.common.sys_storage import FSUtil
+from spark_rapids_pytools.pricing.price_provider import PriceProvider
+
+
+@dataclass
+class DataprocCatalogContainer(JSONPropertiesContainer):
+    def _init_fields(self) -> None:
+        # the prices of the products are defined under 'gcp_price_list'
+        self.props = self.props['gcp_price_list']
+
+
+@dataclass
+class DataprocPriceProvider(PriceProvider):
+    """
+    Provide costs of Dataproc instances
+    """
+    name = 'Dataproc'
+
+    def _process_resource_configs(self):
+        online_entries = self.pricing_config.get_value('catalog', 'onlineResources')
+        for online_entry in online_entries:
+            if online_entry.get('resourceKey') == 'gcloud-catalog':
+                file_name = online_entry.get('localFile')
+                self.cache_file = FSUtil.build_path(self.cache_directory, file_name)
+                self.resource_url = online_entry.get('onlineURL')
+                break
+
+    def _create_catalog(self):
+        self.catalog = DataprocCatalogContainer(prop_arg=self.cache_file)
+
+    def get_ssd_price(self, machine_type: str) -> float:
+        lookup_key = 'CP-COMPUTEENGINE-LOCAL-SSD'
+        return self.catalog.get_value(lookup_key, self.region) * 0.513698630136986
```

Comment:
what does the 0.513... value represent and why is it hard-coded?

Reply:
I moved it into the configurations. Note that I disabled the code of the SSD for now so that our model is consistent across all the platforms. Since we do not include SSD costs in EMR, I disabled the SSD for dataproc as well.

According to GCloud documentation, a single local SSD is fixed size 375GB size.
But for some reason they consider that a single SSD unit is ~750GB. For example, if you experiment with the dataproc pricing calculator, you will find that SSD unit price is $0.08, and a single local SSD is equivalent to 0.513 quantity.